recipes/GKE Setup/README.md

```diff
@@ -0,0 +1,105 @@
+role](https://cloud.google.com/iam/docs/understanding-roles#basic). Else, you will need permission to
+create/edit the following resources:
+* Google Cloud Storage (GCS) buckets
+* BigQuery Datasets
+* GKE Kubernetes objects
+* Service Accounts
+* Service Account IAM Policies
+
+Ensure that you have the [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) and `kubectl` installed
+and that you've [enabled cluster access](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl).
+
+To run the `./deploy.sh` script, you must also have the following CLI tools installed and in your $PATH
+* `gsutil` - Storage CLI (installable with `gcloud components install gsutil`)
+* `bq` - BigQuery CLI (installable with `gcloud components install bq`)
+* [`jq`](https://stedolan.github.io/jq/) - JSON processor
+
+## Prerequisites
+As described in the [cluster access documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl),
+first, ensure that your gcloud SDK is initialized:
+```shell
+gcloud init
+```
+This will authenticate the SDK, allow you to select a project, and configure a default zone/region.
+
+Then (assuming you are using an existing cluster) you must generate a kubernetes configuration file:
+```shell
+gcloud container clusters get-credentials "<cluster-id>"
+```
+Before running the `deploy.sh` script, you must
+[create a BigQuery dataset](https://cloud.google.com/bigquery/docs/datasets#create-dataset) (or use an existing dataset)
+and a [Google Storage bucket](https://cloud.google.com/storage/docs/creating-buckets).
+
+As a default, the script expects the name of the dataset to be `fullstory-export-data` and the
+name of the bucket to be `$GOOGLE_PROJECT-hauser`, but feel free to change these.
+
+If you did not use the default dataset and bucket names, or want to customize the name of the service account,
+feel free to edit the following variables at the top of the `deploy.sh` script.
+```shell
+BIGQUERY_DATASET="fullstory-export-data"
+SERVICE_NAME="hauser-svc"
+GCS_BUCKET="$GOOGLE_PROJECT-hauser"
+HAUSER_VERSION="latest"
+```
+To find your FullStory Org Id, see [help docs](https://help.fullstory.com/hc/en-us/articles/360047075853-How-do-I-find-my-FullStory-Org-Id-).
+You will also need to create or get a [FullStory API Key](https://help.fullstory.com/hc/en-us/articles/360020624834-Where-can-I-find-my-API-key-).
+
+The `HAUSER_VERSION` variable corresponds to the docker image tag. See [docker hub](https://hub.docker.com/r/fullstorydev/hauser) for available tags.
+
+## The deploy script
+The `deploy.sh` script in this directory uses the gcloud SDK and associated CLI's to
+create a service account, add the permissions, and create a Kubernetes deployment that
+runs hauser.
+
+Assuming you have completed the prerequisites, you can run the deploy script with:
+```shell
+./deploy.sh -p "<my_project_id>" -o "<my_org_id>" -k "<fs_api_key"
```

Comment:
It's usually bad practice to stick secrets in command-line args. Perhaps this should accept the API key via env var instead, and then complain at script startup if the env var is unset?

(Also, nit, missing a trailing `>`.)

Reply:
Yeah, I was vacillating on that. It actually already works if you don't provide it. I'll just remove it from the args and require that it's set before running the script.